src/gsi/deter_sfc_mod.f90

```diff
@@ -600,7 +600,7 @@ subroutine deter_sfc2(dlat_earth,dlon_earth,obstime,idomsfc,tsavg,ff10,sfcr,zz)
 
 
 !    Get time interpolation factors for surface files
-     if(obstime > hrdifsfc(1) .and. obstime <= hrdifsfc(nfldsfc))then
+     if(obstime > hrdifsfc(1) .and. obstime < hrdifsfc(nfldsfc))then
```

Comment:
It was changed so that when obstime = hrdifsfc(nfldsfc) it would got to the else code.  Within round-off it should be the same, but the else part of the code has fewer calculations.  This way the obstime<=hrdifsfc(1) operates the same way as obstime >=hrdifsfc(nfldsfc).